mlir-sycl/lib/Transforms/Inliner.cpp

```diff
@@ -360,6 +360,9 @@ class Inliner : public InlinerInterface {
 
   /// The inline heuristic controlling when to inline a call edge.
   const InlineHeuristic &Heuristic;
+
+  /// Whether to inline SYCLMethodOp operations.
+  const bool InlineSYCLMethodOps;
```

Comment:
Would it make sense to have this flag in the InlineHeuristic ?

Reply:
The issue here's that we might not want `alwaysinline` `SYCLMethodOps` to be inlined and we're always inlining those (which I think makes sense)